Tudat/Astrodynamics/Propulsion/costateBasedThrustGuidance.cpp

```diff
@@ -0,0 +1,106 @@
+
+#include <cmath>
+
+#include <boost/function.hpp>
+
+#include <Eigen/Core>
+
+#include "Tudat/SimulationSetup/tudatSimulationHeader.h"
+#include "Tudat/Astrodynamics/Propulsion/costateBasedThrustGuidance.h"
+
+namespace tudat
+{
+
+namespace propulsion
+{
+
+//! Constructor
+MeeCostateBasedThrustGuidance::MeeCostateBasedThrustGuidance(
+        const boost::function< Eigen::Vector6d( ) > thrustingBodyStateFunction,
+        const boost::function< Eigen::Vector6d( ) > centralBodyStateFunction,
+        const boost::function< double( ) > centralBodyGravitationalParameterFunction,
+        boost::function< Eigen::VectorXd( const double ) > costateFunction,
+        const boost::function< Eigen::Vector3d( ) > bodyFixedForceDirection )
+    : BodyFixedForceDirectionGuidance( bodyFixedForceDirection ),
+      thrustingBodyStateFunction_( thrustingBodyStateFunction ),
+      centralBodyStateFunction_( centralBodyStateFunction ),
+      centralBodyGravitationalParameterFunction_( centralBodyGravitationalParameterFunction ),
+      costateFunction_( costateFunction ){ }
+
+//! Function to update the force direction to the current time.
+void MeeCostateBasedThrustGuidance::updateForceDirection( const double time )
+{
+    if( !( time == currentTime_ ) )
+    {
+        Eigen::VectorXd costates_ = costateFunction_( time );
```

Comment:
I would add a check for the length of the costates vector, as it is assumed that it has length 5.